galeracluster/source/selinux.rst

```diff
@@ -0,0 +1,128 @@
+
+In the context of Galera Cluster, systems with SELinux may block the database server, keeping it from starting or preventing the node from establishing connectivity with the cluster.  To avoid this, you need to configure SELinux policies to allow the node to operate.
+
+
+-----------------------------
+Generating an SELinux Policy
+-----------------------------
+.. _`gen-selinux-policy`:
+
+In order to create an SELinux policy for Galera Cluster, you need to first open ports and set SELinux into permissive mode.  Then, generate after generating various replication event, state transfers and notifications, create a policy from this activity, then restore SELinux.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Disabling SELinux for Galera Cluster
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`disable-selinux`:
+
+Building a policy for SELinux requires that you first disable it.  More specifically, set SELinux to open the necessary ports and to permit activities originating from the database server.  While in permissive mode, SELinux continues to create log entries for system events, but takes no action to restrict this activity. 
+
+To disable SELinux, complete the following steps:
+
+#. Using ``semanage``, open the relevant ports:
+
+   .. code-block:: console
+
+      # semanage port -a -t mysqld_port_t -p tcp 4567
+      # semanage port -a -t mysqld_port_t -p tcp 4568
+      # semanage port -a -t mysqld_port_t -p tcp 4444
+
+   SELinux already opens the standard MySQL port ``3306``.  In the event that you use UDP in your cluster, you also need to open ``4567`` to those connections.
+
+   .. code-block:: console
+
+      # semanage port -a -t mysqld_port_t -p udp 4567
+
+#. Set SELinux to permissive mode for the database server.
+
+   .. code-block:: console
+
+      # semanage permissive -a mysqld_t
+
+SELinux now permits the database server to function on the server and no longer blocks the node from network connectivity with the cluster.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Defining the SELinux Policy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`define-selinux-policy`:
+
+While SELinux remains in permissive mode, it continues to log activity from the database server.  In order for it to understand normal operation for the database, you need to start the database and generate routine events for SELinux to see.
+
+For servers that use ``init``, start the database with the following command:
+
+.. code-block:: console
+		
+   # service mysql start
+
+For servers that use ``systemd``, instead run this command:
+
+.. code-block:: console
+
+   # systemctl mysql start
+
+You can now begin to create events for SELinux to log.  There are many ways to go about this, including:
+
+- Deleting the write-set cache forces the node to restore itself using a :term:`State Snapshot Transfer`.  The write-set cache is typically in the datadir and is named ``galera.cache`.
```

Comment:
I do not think this is correct. Will get back to you with the proper way to force SST.